### ISSUE TITLE
Support CLS transaction with multiple databases

### ISSUE THREAD

**Issue by eyalroth** [sequelize/sequelize#14698]
## Issue Creation Checklist

- [x] I understand that my issue will be automatically closed if I don't fill in the requested information
- [x] I have read the [contribution guidelines](https://github.com/sequelize/sequelize/blob/main/CONTRIBUTING.md)

### Describe the feature you'd like to see implemented

I would like to be able to [automatically pass transactions via CLS](https://sequelize.org/docs/v6/other-topics/transactions/#automatically-pass-transactions-to-all-queries) in conjunction with multiple databases (separate transaction per database).

Example:

```js
const animals = new Sequelize(...);
const trees = new Sequelize(...);

animals.useCls(createNamespace('animals'));
trees.useCls(createNamespace('trees'));

await animals.transaction(async () => {
  await trees.transaction(async () => {
    await Dog.create(...);
    await Oak.create(...);
  });
});
```

Currently, this won't work since the CLS transaction namespace is maintained globally, and so the transaction of the second client will override the first one, making all the models/queries of the first client use the second client's transaction instead, which in turn means that it'll use the second client's connection and fail.

### Describe why you would like this feature to be added to Sequelize

I think this self explanatory already.

### Is this feature dialect-specific?

- [x] No. This feature is relevant to Sequelize as a whole.
- [ ] Yes. This feature only applies to the following dialect(s): 

### Would you be willing to resolve this issue by submitting a Pull Request?

- [ ] Yes, I have the time and I know how to start.
- [x] Yes, I have the time but I will need guidance.
- [ ] No, I don't have the time, but my company or I are [supporting Sequelize through donations on OpenCollective](https://opencollective.com/sequelize).
- [ ] No, I don't have the time, and I understand that I will need to wait until someone from the community or maintainers is interested in implementing my feature.

---

*Indicate your interest in the addition of this feature by adding the 👍 reaction. Comments such as "+1" will be removed.*

**Comment by ephys** [comment]
This is something we're planning to do as part of our move to node's built-in AsyncLocalStorage

**Comment by eyalroth** [comment]
@ephys Awesome, that's great to hear 😃

**Comment by nicoabie** [comment]
@ephys If there is an open transaction sequelize will take the connection of that transaction instead of the sequelize instance. 
A check could be added to see if the name of both db connections is the same (or dialect + db name). if it is use the same transaction and if not use another.
Tagging @ivanlolivier who found the issue.

**Comment by ephys** [comment]
Fixed by #15292

**Comment by h4nk2024** [comment]
is this change not available in 6.37.1 ?

**Comment by ephys** [comment]
No that would be a breaking change
